hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -96,7 +106,16 @@ public T getData() {
 
   @Override
   public Comparable<?> getOrderingValue(Schema recordSchema, Properties props) {
-    return this.getData().getOrderingValue();
+    // For non-delete record.
+    if (data != null) {
```

Comment:
The data is designed to be non-null, we even have some validation logic for it in `HoodieRecord`:

```java
public T getData() {
    if (data == null) {
      throw new IllegalStateException("Payload already deflated for record.");
    }
    return data;
  }
```

Reply:
We can check the metadata first in the validation.